### PR TITLE
[SPARK-29008][SQL][FOLLOW-UP] Checks correct param lengths of split methods

### PR DESCRIPTION
**Pull request by maropu**
### What changes were proposed in this pull request?

This pr is to fix wrong code to check parameter lengths of split methods in `subexpressionEliminationForWholeStageCodegen`.

### Why are the changes needed?

Bug fix.

### Does this PR introduce any user-facing change?

No.

### How was this patch tested?

Existing tests.

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala [hunk 1]
```diff
@@ -1050,7 +1050,10 @@ class CodegenContext extends Logging {
     }
 
     val codes = if (commonExprVals.map(_.code.length).sum > SQLConf.get.methodSplitThreshold) {
-      if (commonExprs.map(calculateParamLength).forall(isValidParamLength)) {
+      val inputVarsForAllFuncs = commonExprs.map { expr =>
+        getLocalInputVariableValues(this, expr.head).toSeq
+      }
+      if (inputVarsForAllFuncs.map(calculateParamLengthFromExprValues).forall(isValidParamLength)) {
```
Comment (maropu) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala]: @viirya @cloud-fan sorry, but I found my mistake made in SPARK-29008... Could you check?

**Review by viirya — APPROVED** [review]

**Comment by maropu** [comment]
@viirya many thanks!

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala (hunk 1 above)
Comment (viirya) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala]: yea, good catch! We should check param length for input parameters to common expressions.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM, too. Thank you, @maropu and @viirya .
Merged to master.

**Comment by maropu** [comment]
Thanks for the merging, @dongjoon-hyun !

**Comment by cloud-fan** [comment]
late LGTM, good catch!

**Pull request closed without merging**